### PR TITLE
Task cancelation on job timeout

### PR DESCRIPTION
**Pull request by richtja**
This commit adds a mechanism for terminating running tasks after the job reaches its timeout.

Reference: #5295
Signed-off-by: Jan Richter <jarichte@redhat.com>

### CODE REVIEW

**Review by clebergnu — COMMENTED** [review]
Hi @richtja, this looks quite good, but I have a pending question about `time.monotonic()`.  Thanks.

File: avocado/core/task/statemachine.py [hunk 1]
```diff
@@ -155,6 +160,44 @@ def __repr__(self):
...
+        :param terminate_tasks: runtime_tasks which were terminated
+        :type terminate_tasks: list
+        """
+        for terminated_task in terminate_tasks:
+            task_id = str(terminated_task.task.identifier)
+            job_id = terminated_task.task.job_id
+            log_message = {
+                "status": "running",
+                "type": "log",
+                "log": f"{time.strftime('%Y-%m-%d %H:%M:%S', time.localtime())} | "
```
[the remark this answers is not in the document]
Comment (richtja) [reply avocado/core/task/statemachine.py:178]: I am not sure if I understand your proposal. This is not the message timestamp, but the log, which will be saved to the `debug.log` file. IIUC how `time.monotonic()` works I can not convert it to the human-readable local time, which I need for the logs. That is why I didn't use it here, but maybe I am missing something.

File: avocado/core/task/statemachine.py [hunk 2]
```diff
@@ -155,6 +160,44 @@ def __repr__(self):
...
+                "Runner error occurred: Timeout reached".encode(),
+                "encoding": "utf-8",
```
[the remark this answers is not in the document]
Comment (richtja) [reply avocado/core/task/statemachine.py:180]: Make sense, I will update it. So we won't add such problems into new code.

File: avocado/core/task/statemachine.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (clebergnu) [reply avocado/core/task/statemachine.py:179]: Oh, my mistake, never mind.

**Review by clebergnu — APPROVED** [review]
LGTM, thanks!

**Pull request merged**